### PR TITLE
Add Jellyfin support, web client (movies/tv) and official website.

### PR DESCRIPTION
**Pull request by alexbcberio**
Add Jellyfin presence for both official site [jellyfin.org](https://jellyfin.org) and self hosted site (web client).

The official site presence if fully finished.
For the web client, movies and tv show type libraries have full support as well as any other web panels.

### CODE REVIEW

**Comment by doomlerd** [comment]
You need to provide screenshots as proof that the presence works.

**Comment by alexbcberio** [comment]
Screenshot of presence working:

#### TvShow Playing
TvShow-Playing
#### TvShow Paused
TvShow-Paused
#### Movie Playing
Movie-Playing
#### Browsing details of Movie
BrowsingDetails-Movie

Feel free to contact for any other details.

**Comment by alexbcberio** [comment]
I forgot to uncomment resetting the presence default values, so when exiting the playback the small icon was kept.

File: Jellyfin/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    "id": "202915432175239169"
+  },
+  "url": "jellyfin.org",
+  "description": {
+    "en": "The Free Software Media System"
+  },
+  "service": "Jellyfin",
+  "version": "1.0",
+  "logo": "https://raw.githubusercontent.com/jellyfin/jellyfin/master/Emby.Dlna/Images/logo240.png",
+  "thumbnail": "https://jellyfin.org/favicon.ico",
```
Comment (Bas950) [inline Jellyfin/dist/metadata.json]: Don't use github's cdn or their own, please upload the images to a site like imgur.

File: Jellyfin/dist/metadata.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  "color": "#00698c",
+  "category": "videos",
+  "tags": [
+      "video",
+      "movie",
+      "tv",
+      "media"
+  ],
+  "regExp": "([a-z0-9]+[.])*([a-z0-9]+)"
```
Comment (Bas950) [inline Jellyfin/dist/metadata.json]: Please make sure other presences still work, because you put *

File: Jellyfin/dist/presence.js [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+    // details: "",
+
+    // The lower section of the presence text, 3rd line
+    // state: "",
+
+    // The unix epoch timestamp for when to start counting from
+    // startTimestamp: null,
+
+    // If you want to show Time Left instead of Elapsed, this is the unix epoch timestamp at which the timer ends
+    // endTimestamp: 1577151472000
```
Comment (Bas950) [inline Jellyfin/dist/presence.js]: Please remove all the comments that are from the docs, please leave only comments about the presence itself or your todos etc.

File: Jellyfin/dist/metadata.json (hunk 1 above)
Comment (doomlerd) [reply Jellyfin/dist/metadata.json]: the thumbnail is also a favicon xd, it should be an actual wide +500px image

Comment (alexbcberio) [reply Jellyfin/dist/metadata.json]: Updated in commit [5291b2347b892984d9236c93f73282aca2f9c483](https://github.com/alexbcberio/Presences/commit/5291b2347b892984d9236c93f73282aca2f9c483)

File: Jellyfin/dist/presence.js (hunk 3 above)
Comment (alexbcberio) [reply Jellyfin/dist/presence.js]: Comments from the docs removed in commit [cdd7d3f7fec4a363fa028dd01fa3fabbe5d2bd76](https://github.com/alexbcberio/Presences/commit/cdd7d3f7fec4a363fa028dd01fa3fabbe5d2bd76)

File: Jellyfin/dist/metadata.json (hunk 2 above)
Comment (alexbcberio) [reply Jellyfin/dist/metadata.json]: Other presences worked previously Ok, I've changed the way the presence is initialized in [845dbfeb8bfd17c1cc4d9b7543e16ddcd99ec05e](https://github.com/alexbcberio/Presences/commit/845dbfeb8bfd17c1cc4d9b7543e16ddcd99ec05e).

In the new commit the presence is only initialized if we are in the official page or we are in the web client, by accessing a variable on the localstorage that keeps the last log in timestamp and verifying that it has been in the previous 30 seconds.

File: Jellyfin/dist/presence.js [hunk 4]
```diff
@@ -0,0 +1,459 @@
...
+            clientId: "669359568391766018",
+            mediaKeys: false
+        });
+
+        presence.on("UpdateData", updateData);
+    }
+}
+init();
+
+// View https://docs.premid.app/en/dev/presence/class -> presenceData interfac
```
Comment (Bas950) [inline Jellyfin/dist/presence.js]: Can be removed.

File: Jellyfin/dist/presence.js [hunk 5]
```diff
@@ -0,0 +1,459 @@
...
+let presenceData = {
+    largeImageKey: PRESENCE_ART_ASSETS.logo,
+    // smallImageKey: "key",
+    // smallImageText: "Some hover text",
+    // details: "",
+    // state: "",
+    // startTimestamp: null,
+    // endTimestamp: 1577151472000
```
Comment (Bas950) [inline Jellyfin/dist/presence.js]: remove comma after large image key, and remove the unneeded comments.

File: Jellyfin/dist/presence.js (hunk 4 above)
Comment (Bas950) [reply Jellyfin/dist/presence.js]: (line 60)

File: Jellyfin/dist/presence.js (hunk 5 above)
Comment (alexbcberio) [reply Jellyfin/dist/presence.js]: Removed comments in commit [2cbff2aadb8940809157cd9169073603a8124af7](https://github.com/alexbcberio/Presences/commit/2cbff2aadb8940809157cd9169073603a8124af7)

**Comment by alexbcberio** [comment]
Proofs of working presence:
#### Official website
jellyfin org
#### Playing a series
Playing series
#### Playing a series (paused state)
paused series
#### Playing a movie
playing movie
#### Browsing details of a movie
browsing details

**Pull request merged**